ws-client/src/client.rs

```diff
@@ -455,6 +456,42 @@ impl SubscriptionClient for WsClient {
 		};
 		Ok(Subscription { to_back: self.to_back.clone(), notifs_rx, marker: PhantomData, id })
 	}
+
+	/// Register a notification handler for async messages from the server.
+	///
+	async fn on_notification<'a, N>(&self, method: &'a str) -> Result<Subscription<N>, Error>
```

Comment:
I think we need a new type for `Subscription`, the drop implementation will not work I think.

Reply:
Yeah agreed actually this would let me clean things up quite a bit. I'll work on that today